scripts/unified_state_manager.py

```
@@ -0,0 +1,632 @@
+            Lock object when acquired
+        """
+        import os
+        timeout = timeout or self.LOCK_TIMEOUT
+
+        if HAS_FILELOCK:
+            lock = filelock.FileLock(str(self.lock_file), timeout=timeout)
+            try:
+                with lock.acquire(timeout=timeout):
+                    yield lock
+            except filelock.Timeout:
+                raise TimeoutError(f"Could not acquire lock within {timeout} seconds")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)